OpenRA.Mods.Common/Traits/Attack/AttackBase.cs

```diff
@@ -223,13 +223,29 @@ public IEnumerable<Armament> ChooseArmamentsForTarget(Target t, bool forceAttack
 			// (short-circuiting in the logical expression below)
 			var owner = null as Player;
 			if (t.Type == TargetType.FrozenActor)
-				owner = t.FrozenActor.Owner;
+			{
+				owner = t.FrozenActor.Actor.EffectiveOwner != null && t.FrozenActor.Actor.EffectiveOwner.Owner != null
```

Comment:
This will crash if the actor is dead (`t.FrozenActor.Actor == null`).  You need to check for that case and use `t.FrozenActor.Owner` instead.